### ISSUE TITLE
Clean out Travis build matrix to speed up CI

### ISSUE THREAD

**Issue by jesperronn** [jmazzi/crypt_keeper#114]
This innocent looking support matrix in [.travis.yml](https://github.com/jmazzi/crypt_keeper/blob/master/.travis.yml#L3)  is very deceptive
```
rvm:
  - 1.9.3
  - 2.0.0
  - 2.1.1

gemfile:
  - gemfiles/activerecord_3_1.gemfile
  - gemfiles/activerecord_3_2.gemfile
  - gemfiles/activerecord_4_0.gemfile
  - gemfiles/activerecord_4_1.gemfile
  - gemfiles/activerecord_4_2.gemfile

matrix
  include:
    - rvm: 2.2.0
      gemfile: gemfiles/activerecord_4_2.gemfile
```

this runs 3x5+1 = 16 jobs and takes 21 minutes before you get feedback after commit.

We also need to add newer ruby versions: 2.3.0 at least, and the upcoming rails 5 version.

Any suggestions as to which jobs should be weeded out to make space for newer versions?

**Comment by tfwright** [comment]
Ruby less that 2.1 is EOL I believe, so those can probably be removed. I think it's probably also fine to remove AR 3x.

**Comment by jesperronn** [comment]
Good point:)

**Comment by itspriddle** [comment]
I think its fine to remove ActiveRecord 4.0 as well. It doesn't seem it is receiving updates any more: http://guides.rubyonrails.org/maintenance_policy.html

Ruby 2.1 is _near_ EOL, and only receiving security fixes: https://www.ruby-lang.org/en/news/2016/02/24/support-plan-of-ruby-2-0-0-and-2-1/

I'd propose the next CryptKeeper release remove formal support for these and test the following:

- Latest Ruby 2.2
- Latest Ruby 2.3
- Latest ActiveRecord 4.1
- Latest ActiveRecord 4.2
- Latest ActiveRecord 5.0

**Comment by jesperronn** [comment]
I follow you completely, will make a PR for you to merge

**Comment by jmazzi** [comment]
@jesperronn thanks for the proposal! Sorry for the delay in responding. The changes have been merged in #118

**Comment by jesperronn** [comment]
Awesome thanks!
